packages/odyssey-react-mui/src/createContrastColors.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { DesignTokens } from "./OdysseyDesignTokensContext";
+
+type RgbColorObject = {
+  red: number;
+  green: number;
+  blue: number;
+};
+
+export type ContrastColors = {
+  focusRingColor: string | undefined;
+  fontColor: string | undefined;
+  itemDisabledFontColor: string | undefined;
+  itemHoverBackgroundColor: string | undefined;
+  itemSelectedBackgroundColor: string | undefined;
+};
+
+const hexToRgb = (hexBackgroundColor: string): RgbColorObject | undefined => {
+  const formattedHexString = hexBackgroundColor.includes("#")
+    ? hexBackgroundColor.split("#")[1]
+    : hexBackgroundColor;
+
+  return {
+    red: parseInt(formattedHexString.slice(0, 2), 16),
+    green: parseInt(formattedHexString.slice(2, 4), 16),
+    blue: parseInt(formattedHexString.slice(4, 6), 16),
+  };
+};
+
+export const generateContrastColors = (
+  backgroundColor: string,
+  odysseyDesignTokens: DesignTokens,
+) => {
+  // Convert hex to RGB
+  const rgbFromHex = hexToRgb(backgroundColor);
+
+  if (rgbFromHex) {
+    const { red, green, blue } = rgbFromHex;
+
+    // Calculate relative luminance
+    // @see https://contrastchecker.online/color-relative-luminance-calculator#:~:text=For%20the%20sRGB%20colorspace%2C%20the,%2B0.055)%2F1.055)%20%5E%202.4
+    // returns a number between 0(black) and 255(white)
+    const luminance = 0.2126 * red + 0.7152 * green + 0.0722 * blue;
+
+    // 128 is a magic number. This feels like roughly where we should switch from dark to light.
+    const LUMINANCE_THRESHOLD = 128;
+    const LUMINANCE_EDGE_MIN = 108;
+    const LUMINANCE_EDGE_MAX = 142;
+
+    // Luminance values between LUMINANCE_EDGE_MIN-LUMINANCE_EDGE_MAX can cause contrast ration issues
+    // Using #000000 helps in these cases
+    const luminanceValueInEdgeRange =
+      luminance > LUMINANCE_EDGE_MIN && luminance < LUMINANCE_EDGE_MAX;
+
+    // Determine if the color is light or dark.
+    const isLight = luminance > LUMINANCE_THRESHOLD;
+
+    const fontColor = luminanceValueInEdgeRange
+      ? "#000000"
```

Comment:
Can we make this a const? Same with white, we should probably just set those ourselves in this file rather than using design tokens because we're forcing that for contrast reasons.